src/mongodb/scores.controller.ts

```diff
@@ -699,137 +1572,1444 @@ export class ScoresController {
+        if (totalSessionTargets >= 30) {
+          sessionLevel = 1;
+        } else if (totalSessionTargets < 30 && totalSessionTargets >= 16) {
+          console.log(totalSessionTargets);
+          console.log(totalSessionValidations);
+          if (totalSessionValidations > 5) {
+            sessionLevel = 1;
+          } else {
+            sessionLevel = 2;
+          }
+        } else if (totalSessionTargets < 16 && totalSessionTargets > 2) {
+          if (totalSessionValidations > 2) {
+            sessionLevel = 2;
+          } else {
+            sessionLevel = 3;
+          }
+        } else if (totalSessionTargets <= 2) {
+          if (totalSessionValidations > 0) {
+            sessionLevel = 3;
+          } else {
+            sessionLevel = 4;
+          }
+        }
+      }
+
+      if (sessionHighestLevel < sessionLevel) {
+        sessionHighestLevel = sessionLevel;
+        sessionHighestTargets = totalSessionTargets;
+        sessionHighestValidations = totalSessionValidations;
+      }
+    }
+
+    console.log(sessionHighestLevel);
+
+    if (currentLevel < sessionHighestLevel) {
+      currentLevel = sessionHighestLevel;
+      totalTargets = sessionHighestTargets;
+      totalValidation = sessionHighestValidations;
+    }
+
+    currentLevel = 'm' + currentLevel;
+
+    return { currentLevel: currentLevel, targetsCount: totalTargets, validationsCount: totalValidation, targets: targets, validations: validations };
+  }
+
+  @ApiExcludeEndpoint(true)
+  @Post('/getSetResult')
+  async getSetResult(@Res() response: FastifyReply, @Body() getSetResult: any) {
+
+    let targets = await this.scoresService.getTargetsBysubSession(getSetResult.sub_session_id, getSetResult.contentType);
+
+    let totalTargets = targets.length;
+    let sessionResult = 'No Result';
+
+    let recordData: any = await this.scoresService.getlatestmilestone(getSetResult.user_id);
+    let previous_level = recordData[0]?.milestone_level || undefined;
+
+    if (getSetResult.contentType === 'Char' || getSetResult.contentType === 'char') {
+      if (totalTargets < 5) {
+        sessionResult = 'pass';
+      } else {
+        sessionResult = 'fail';
+      }
+    } else if (getSetResult.contentType === 'Word' || getSetResult.contentType === 'word') {
+      if (totalTargets < 15) {
+        sessionResult = 'pass';
+      } else {
+        sessionResult = 'fail';
+      }
+    } else if (getSetResult.contentType === 'Sentence' || getSetResult.contentType === 'sentence') {
+      if (totalTargets < 15) {
+        sessionResult = 'pass';
+      } else {
+        sessionResult = 'fail';
+      }
+    } else if (getSetResult.contentType === 'Paragraph' || getSetResult.contentType === 'paragraph') {
+      if (totalTargets < 15) {
+        sessionResult = 'pass';
+      } else {
+        sessionResult = 'fail';
+      }
+    }
+
+    if (previous_level === undefined || getSetResult.collectionId === "5221f84c-8abb-4601-a9d0-f8d8dd496566") {
```

Comment:
Collection id cannot be relied upon.  Please use the content Id to extract the type of content it is and use that to evaluation.
For example for ASER and other Assessment related content, you can add extra attribute that would help you to identify here.